src/widgets/ArticleBlock/ui/ArticleBlock.tsx

```diff
@@ -0,0 +1,57 @@
+import { FC, useState } from 'react'
+
+import description from '@/assets/images/articleBlock/description.png'
+import ArrowIcon from '@/assets/images/sideBarMenu/IconArrowDown.svg'
+import { Button } from '@/shared/ui/Button/Button'
+import Heading, { HeadingType } from '@/shared/ui/Heading/Heading'
+import Paragraph from '@/shared/ui/Paragraph/Paragraph'
+
+import styles from './articleBlock.module.scss'
+
+/**
+ * Компонент ArticleBlock используется как виджет на главной странице MainPage
+ */
+
+const ArticleBlock: FC = () => {
+  const [isActive, setIsActive] = useState(false)
+  function openTextHandler() {
+    setIsActive(!isActive)
```

Comment:
Когда мы ориентируемся на прошлое состояние state - то передаем в функцию колбек
```suggestion
    setIsActive(prev => !prev)
```